packages/pwa/app/components/cart-quantity-picker/index.js

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useState} from 'react'
+import PropTypes from 'prop-types'
+import {HStack, Button, Input, useDisclosure} from '@chakra-ui/react'
+import {useProduct} from '../../hooks'
+import ConfirmationModal from '../confirmation-modal'
+import {REMOVE_CART_ITEM_CONFIRMATION_DIALOG_CONFIG} from '../../pages/cart/partials/cart-secondary-button-group'
+
+const CartQuantityPicker = ({product, handleRemoveItem, onItemQuantityChange}) => {
+    const {stepQuantity} = useProduct(product)
+    const [updatedQuantity, setUpdatedQuantity] = useState(product.quantity)
+    const modalProps = useDisclosure()
+
+    const showRemoveItemConfirmation = () => {
+        modalProps.onOpen()
+    }
+    return (
+        <HStack maxW="80%">
+            <Button
+                data-testid="cart-quantity-decrement"
+                variant="outline"
+                onClick={() => {
+                    if (product.quantity - stepQuantity === 0) {
+                        showRemoveItemConfirmation()
+                    } else {
+                        setUpdatedQuantity(product.quantity - stepQuantity)
+                        onItemQuantityChange(product.quantity - stepQuantity)
+                    }
+                }}
+            >
+                -
+            </Button>
+            <form
+                onSubmit={(e) => {
+                    if (parseInt(updatedQuantity)) {
+                        onItemQuantityChange(updatedQuantity)
+                    } else if (updatedQuantity === '0') {
+                        setUpdatedQuantity(0)
+                        showRemoveItemConfirmation()
+                    }
+                    e.preventDefault()
+                    document.activeElement.blur()
+                }}
+            >
+                <Input
```

Comment:
It would be super neato if the text would be selected when you click or tap on the input. You can see how that is done [here](https://stackoverflow.com/questions/6139107/programmatically-select-text-in-a-contenteditable-html-element).

Reply:
You might want to confer with @jennmills on what the width of this input should be, right now its only wide enough for double digits. 3 might be safer choice.